src/main.rs

```diff
@@ -55,36 +55,25 @@ async fn main() -> Result<(), failure::Error> {
 
     debug!("Got config as {:?}", &leaked_config);
 
-    let tl_t: JoinHandle<Result<(), Error>> =
-        tokio::spawn(async move { 
-            loop {
-                info!("going to connect (monitor) in loop...");
-                match torrentleech::monitor(&leaked_config.torrentleech).await {
-                    Ok(_) => {
-                        error!("monitor resolved w/ Ok(), should be impossible!");
-                        return Ok(());
-                    },
-                    Err(e) => match e.downcast_ref::<irc::error::Error>() {
-                        Some(irc::error::Error::PingTimeout) => {
-                            error!("got a ping timeout! will try and reconnect")
-                        },
-                        Some(other) => {
-                            error!("Got some other IRC error: {:?}", other);
-                            return Err(e);
-                        }
-                        None => {
-                            error!("Got non-irc error: {:?}", e);
-                            return Err(e);
-                        }
-                    },
-                }
-            }
-         });
+    let tl_t = monitor_wrapper(torrentleech::monitor, &leaked_config.torrentleech);
+    let ipt_t = monitor_wrapper(ipt::ipt_monitor, &leaked_config.ipt);
+
+    // We don't care about the result (should we?)
+    let (torrentleech_join_result, ipt_join_result) = join!(tl_t, ipt_t);
+    error!("We joined the threads, something went wrong!\nTorrentleech: {:?}\nIPT: {:?}", torrentleech_join_result, ipt_join_result);
+
+    Ok(())
+}
 
-    let ipt_t = tokio::spawn(async move { 
+async fn monitor_wrapper<F, Fut, T: Sync>(monitor_fn: F, config: &'static T) -> tokio::task::JoinHandle<Result<(), failure::Error>> 
+where 
+    F: Fn(&'static T) -> Fut + Send + 'static,
+    Fut: Future<Output = Result<(), failure::Error>> + Send
+{
+    tokio::spawn(async move {
```

Comment:
This was pretty hard to figure out. Basically adding on the `+ Send` stuff manually, and also the `+ 'static` and whatnot.